### ISSUE TITLE
Cannot open IINA when there is no internet

### ISSUE THREAD

**Issue by xgrei** [iina/iina#2907]
![iina](https://user-images.githubusercontent.com/57981640/80692545-e9134500-8afb-11ea-97cb-57ec5e043670.gif)

As you can see in the gif.

IINA version: 1.0.6 Build 113

**Comment by JadedMagician** [comment]
I've also run into this issue. I've been working from home for quarantine and have to use a work VPN on my Mac that blocks external connections. If I try to open a file in IINA while on the VPN, it bounces for several miuntes in the dock before finally either opening the video or quitting.

I'm running 1.0.4 Build 107

**Comment by lhc70000** [comment]
@xgrei If it's indeed caused by the Internet connection, it should be related to Sparkle. IINA won't make network requests except for searching online subtitles. However I can't reproduce this problem if I just unplug my ethernet cable and disable the Wi-Fi. Is there anything else I need to do?

@JadedMagician This is even stranger. IINA won't try to connect to the Internet when opening a file unless it's a remote one. Did you turn on the "search subtitles automatically" option?

**Comment by saagarjha** [comment]
Just IINA? This might be notarization as well…

**Comment by lhc70000** [comment]
Another user reported a similar problem (not sure if it's the same one) and here are some findings:

- The user cannot launch IINA through GUI, but invoking it using the command line (`/Applications/IINA.app/Contents/MacOS/IINA`) works
- According to the sample provided, seems that it got stuck when reading recent documents
- The user can open IINA normally after clearing all playback history and recent documents (Pref - Utils - Clear Playback History).

Process sample:

```
Heaviest stack for the main thread of the target process:
  32  start + 1 (libdyld.dylib + 109769) [0x7fff67fe8cc9]
  32  main + 9 (IINA + 28473) [0x1020c7f39]
  32  NSApplicationMain + 777 (AppKit + 15238) [0x7fff2b224b86]
  32  -[NSApplication run] + 658 (AppKit + 204158) [0x7fff2b252d7e]
  32  -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1352 (AppKit + 262256) [0x7fff2b261070]
  32  _DPSNextEvent + 883 (AppKit + 268329) [0x7fff2b262829]
  32  _BlockUntilNextEventMatchingListInModeWithFilter + 64 (HIToolbox + 193913) [0x7fff2cc1a579]
  32  ReceiveNextEventCommon + 584 (HIToolbox + 194517) [0x7fff2cc1a7d5]
  32  RunCurrentEventLoopInMode + 292 (HIToolbox + 195261) [0x7fff2cc1aabd]
  32  CFRunLoopRunSpecific + 462 (CoreFoundation + 532174) [0x7fff2dfebece]
  32  __CFRunLoopRun + 1871 (CoreFoundation + 535674) [0x7fff2dfecc7a]
  32  __CFRunLoopDoTimers + 322 (CoreFoundation + 646871) [0x7fff2e007ed7]
  32  __CFRunLoopDoTimer + 859 (CoreFoundation + 648175) [0x7fff2e0083ef]
  32  __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__ + 20 (CoreFoundation + 649353) [0x7fff2e008889]
  32  __NSFireTimer + 67 (Foundation + 783082) [0x7fff306e32ea]
  32  @objc AppDelegate.checkForShowingInitialWindow() + 40 (IINA + 997592) [0x1021b48d8]
  32  AppDelegate.showWelcomeWindow(checkingForUpdatedData:) + 289 (IINA + 997905) [0x1021b4a11]
  32  -[NSWindowController showWindow:] + 36 (AppKit + 2855490) [0x7fff2b4da242]
  32  -[NSWindowController window] + 110 (AppKit + 2320260) [0x7fff2b457784]
  32  -[NSWindowController _windowDidLoad] + 624 (AppKit + 2337230) [0x7fff2b45b9ce]
  32  @objc InitialWindowController.windowDidLoad() + 24 (IINA + 19000) [0x1020c5a38]
  32  InitialWindowController.windowDidLoad() + 1419 (IINA + 18555) [0x1020c587b]
  32  -[NSTableView setDataSource:] + 338 (AppKit + 1176096) [0x7fff2b340220]
  32  -[NSTableView _tileAndRedisplayAll] + 207 (AppKit + 991904) [0x7fff2b3132a0]
  32  -[NSView setNeedsDisplayInRect:] + 767 (AppKit + 395918) [0x7fff2b281a8e]
  32  -[NSView(NSInternal) _setLayerNeedsDisplayInViewRect:] + 171 (AppKit + 411617) [0x7fff2b2857e1]
  32  -[NSTableView bounds] + 100 (AppKit + 990851) [0x7fff2b312e83]
  32  -[NSTableView tile] + 285 (AppKit + 992813) [0x7fff2b31362d]
  32  -[NSTableView _minimumFrameSize] + 73 (AppKit + 937332) [0x7fff2b305d74]
  32  -[NSTableView _totalHeightOfTableView] + 274 (AppKit + 938054) [0x7fff2b306046]
  32  -[_NSTableRowHeightStorage numberOfRows] + 38 (AppKit + 939104) [0x7fff2b306460]
  32  -[NSTableView _uncachedNumberOfRows] + 295 (AppKit + 1174047) [0x7fff2b33fa1f]
  32  @objc InitialWindowController.numberOfRows(in:) + 47 (IINA + 22127) [0x1020c666f]
  32  specialized InitialWindowController.numberOfRows(in:) + 97 (IINA + 37233) [0x1020ca171]
  32  -[NSDocumentController(NSPrivate) _recentDocumentURLsForKey:] + 231 (AppKit + 6231648) [0x7fff2b812660]
  32  -[NSOperation waitUntilFinished] + 697 (Foundation + 331467) [0x7fff30674ecb]
  32  __psynch_cvwait + 10 (libsystem_kernel.dylib + 14466) [0x7fff6812c882]
 *32  psynch_cvcontinue + 0 (pthread + 18722) [0xffffff7f82b30922]
```

**Comment by saagarjha** [comment]
Can you grab a spindump of the system while that's occurring, so we can see what the lock is waiting on?

**Comment by riobard** [comment]
I'm running into a similar issue with IINA version 1.3.5. The app icon keeps bouncing at the dock until a system popup warning about connection to my NAS cannot be established (see below). Since the NAS has been decommissioned I cannot reconnect it anymore. I believe IINA got stuck at trying to enumerate recently opened files? 

<img width="282" alt="image" src="https://github.com/iina/iina/assets/22984/bc6144c3-d788-4e29-82ba-d5101e3880ee">

The trick mentioned by @lhc70000 above by launching directly from `/Applications/IINA.app/Contents/MacOS/IINA` does not work either. 

Is there anyway to make IINA launch from a clean state (i.e. without enumerating recently opened files list)?

**Comment by lhc70000** [comment]
@riobard This is issue #4973.

@low-batt We probably need a hotfix since it's reported multiple times. I will have a look this evening.

**Comment by low-batt** [comment]
This issue from 2020 may have been that macOS problem where it took a very long time to start apps if the internet was not available because it was trying to validate certificates with Apple servers. We probably should close this issue.

@riobard As @lhc70000 indicated this is most likely issue #4973. To be sure of our understanding I have some questions. Lets continue this discussion in #4973.

@lhc70000 there is a proposed fix for #4973 ready for review.

**Comment by lhc70000** [comment]
Yes, let's move to 4973. Closing this issue.
